enzy_htp/structure/structure_region/api.py

```diff
@@ -61,10 +63,25 @@ def atoms(self) -> List[Atom]:
         return self.atoms_
 
     @atoms.setter
-    def atoms(self, val: List[Atom]):
+    def atoms(self, val: List[Atom]) -> None:
         """setter for atoms_"""
         self.atoms_ = val
 
+        for atom in self.atoms:
+            if atom.parent.parent is None:
+                atom.parent.parent = self
+        #TODO(CJ): set the parent.
+
+    def add_atoms(self, new_atoms:List[Atom]) -> None:
+        #TODO(CJ): documentation!
+
+        self.atoms.extend(new_atoms)
+
+        for atom in self.atoms:
+            if atom.parent.parent is None:
+                atom.parent.set_parent(self)               
```

Comment:
same as above

Reply:
Fixed using your suggested trick! Thank you. I decided it was best to make a `ResidueCap.root()` method that redirects to the `ResidueCap.link_atom.root()`